core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -994,8 +994,10 @@ protected Scope visitTableExecute(TableExecute node, Optional<Scope> scope)
             TableHandle tableHandle = redirection.getTableHandle()
                     .orElseThrow(() -> semanticException(TABLE_NOT_FOUND, table, "Table '%s' does not exist", tableName));
 
-            // TODO add access check
-            // accessControl.check...(session.toSecurityContext(), tableName);
+            accessControl.checkCanExecuteTableProcedure(
+                    session.toSecurityContext(),
+                    tableName,
+                    createQualifiedObjectName(session, node.getProcedureName(), QualifiedName.of(procedureName)));
```

Comment:
procedure should be identified by a `String`  (just like `io.trino.spi.connector.ConnectorMetadata#getTableHandleForExecute`)


`createQualifiedObjectName` creates `catalog.schema.procudre_name` which has little sense (especially the schema part)